src/core_plugins/region_map/public/region_map_visualization.js

```diff
@@ -49,7 +49,12 @@ export function RegionMapsVisualizationProvider(Private, Notifier, config) {
         return;
       }
 
-      this._updateChoroplethLayer(this._vis.params.selectedLayer.url, this._vis.params.selectedLayer.attribution);
+      this._updateChoroplethLayerForNewMetrics(
```

Comment:
can you quickly explain why all this calls to updateChoroplethLayer ? (4 calls to updateChoroplethLayerForXXX)

Reply:
After  looking into this more in-depth, I can't really trigger this being called 4 times in a single `render` call.
 
This may get called 2 times in a single render call. Once when there is a data change

These checks arose due to a limitation of leaflet, that doesn't allow us to change the data after we have assigned it to a layer. That's why we need to check if we can reuse the existing layer or create a new instance. Whether we can reuse it depends on a number of factors:
- if any parameters change that affect the join-parameters, we cannot reuse the instance
- if the data changes, we cannot reuse the instance.

I think it is working as intended. The checks avoid recreating an instance of the leaflet-layer and downloading new data when we shouldn't. This is an important feature. If we remove this and redraw the visualization from scratch for each render call, this will absolutely kill responsiveness of the map.